client.go

```diff
@@ -69,43 +67,28 @@ func (c *Client) GetBaseUrl() string {
 
 // Validate checks if the API and application keys are valid.
 func (client *Client) Validate() (bool, error) {
-	var bodyreader io.Reader
 	var out valid
-	uri, err := client.uriForAPI("/v1/validate")
-	if err != nil {
-		return false, err
-	}
-	req, err := http.NewRequest("GET", uri, bodyreader)
+	var resp *http.Response
 
-	if err != nil {
+	if uri, err := client.uriForAPI("/v1/validate"); err != nil {
 		return false, err
 	}
-	if bodyreader != nil {
-		req.Header.Add("Content-Type", "application/json")
-	}
 
-	var resp *http.Response
-	resp, err = client.HttpClient.Do(req)
+	req, err := http.NewRequest("GET", uri, nil)
+
+	resp, err = client.doRequestWithRetries(req, client.RetryTimeout)
 	if err != nil {
 		return false, err
 	}
 
-	defer resp.Body.Close()
-
-	// Only care about 200 OK or 403 which we'll unmarshal into struct valid. Everything else is of no interest to us.
-	if resp.StatusCode != 200 && resp.StatusCode != 403 {
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return false, err
-		}
-		return false, fmt.Errorf("API error %s: %s", resp.Status, body)
+	if body, err := ioutil.ReadAll(resp.Body); err != nil {
+		return false, err
 	}
 
-	body, err := ioutil.ReadAll(resp.Body)
-	err = json.Unmarshal(body, &out)
-	if err != nil {
+	if err = json.Unmarshal(body, &out); err != nil {
 		return false, err
 	}
 
+	defer resp.Body.Close()
```

Comment:
Please move the defer up to just after checking if `doRequestWithRetries` return an error, I think below line 82.